lib/fdotypes.c

```diff
@@ -0,0 +1,5652 @@
+}
+
+/**
+ * Free an fdo_string_t object, fdo_free any contained buffer as well
+ * @param b - the fdo_string_t object to be fdo_freed
+ * @return none
+ */
+void fdo_string_free(fdo_string_t *b)
+{
+	if (b) {
+		fdo_string_init(b);
+		fdo_free(b);
+	}
+}
+
+/**
+ * The same as FDOString_empty
+ * @param b - the object to have its buffers fdo_freed
+ * @return pointer to the empty FDOString object
+ */
+void fdo_string_init(fdo_string_t *b)
+{
+	if (b->bytes) {
+		fdo_free(b->bytes);
+		b->bytes = NULL;
+	}
+	b->byte_sz = 0;
+}
+
+/**
+ * Resize the buffer in a fdo_string_t to the new size and
+ * return the space filled with zeros
+ * fdo_free any already present buffers
+ * @param b - the fdo_string_t object to be resized
+ * @param byte_sz - the number of bytes to allocate for the new buffer
+ * @return true if successful, false otherwise
+ */
+bool fdo_string_resize(fdo_string_t *b, int byte_sz)
+{
+	if (!b)
+		return false;
+
+	fdo_string_init(b);
+	if (byte_sz > 0) {
+		b->byte_sz = byte_sz;
+		b->bytes = fdo_alloc(byte_sz * sizeof(char));
+		if (b->bytes)
+			return true;
+		else
+			return false;
+	}
+	return true;
+}
+
+/**
+ * Resize the buffer in a fdo_string_t to the new size and
+ * return the space filled with zeros
+ * fdo_free any already present buffers
+ * @param b - the fdo_string_t object to be resized
+ * @param new_byte_sz - the number of bytes to allocate for the new buffer
+ * @param data - the non zero terminated string to copy
+ * @return true if successful, false otherwise
+ */
+bool fdo_string_resize_with(fdo_string_t *b, int new_byte_sz, const char *data)
+{
+	if (!b || !data)
+		return NULL;
+
+	if (fdo_string_resize(b, new_byte_sz + 1)) {
+		if (new_byte_sz > 0)
+			if (memcpy_s(b->bytes, new_byte_sz, data,
+				     new_byte_sz) != 0) {
+				LOG(LOG_ERROR, "Memcpy Failed\n");
+				fdo_free(b->bytes);
+				return false;
+			}
+
+		return true;
+	} else
+		return false;
+}
+
+/**
+ * Write the SigInfo of the form:
+ * SigInfo = [
+ *   sgType: DeviceSgType,
+ *   Info: bstr
+ * ]
+ * @param fdow - pointer to the struct where the GID is to be written.
+ * @return true if write is successfull. false, otherwise.
+ */
+bool fdo_siginfo_write(fdow_t *fdow)
+{
+	bool ret = false;
+	if (!fdow_start_array(fdow, 2)) {
+		LOG(LOG_ERROR, "SigInfo: Failed to start array\n");
+		return ret;
+	}
+	if (!fdow_unsigned_int(fdow, FDO_PK_ALGO)) {
+		LOG(LOG_ERROR, "SigInfo: Failed to write sgType\n");
+		return ret;
+	}
+
+	fdo_byte_array_t *empty_byte_array = fdo_byte_array_alloc(0);
+	if (!empty_byte_array) {
+		LOG(LOG_ERROR, "SigInfo: Byte Array Alloc failed\n");
+		return false;
+	}
+
+	if (!fdow_byte_string(fdow, empty_byte_array->bytes, empty_byte_array->byte_sz)) {
+		LOG(LOG_ERROR, "SigInfo: Failed to write Info\n");
+		goto end;
+	}
+
+	if (!fdow_end_array(fdow)) {
+		LOG(LOG_ERROR, "SigInfo: Failed to end array\n");
+		goto end;
+	}
+	LOG(LOG_DEBUG, "eASigInfo write successfull\n");
+	ret = true;
```

Comment:
Remove debug log, try to add a generic message for all error log and debug log.
As it may disclose info.